Applications/WebPortal/WebApp/tests/profile_page_tester/source/index.tsx

```diff
@@ -68,11 +83,20 @@ class TestApp extends React.Component<{}, State> {
     }
   }
 
-  private onInput(value: string) {
+  private onTextInput(value: string) {
     this.setState({
       lastNameValue: value
     });
   }
+
+  private onRoleClick(role: Nexus.AccountRoles.Role) {
+    if(this.state.someRoles.test(role)) {
+      this.state.someRoles.unset(role);
+    } else {
+      this.state.someRoles.set(role);
+    }
+    this.forceUpdate();
```

Comment:
Instead of ```forceUpdate``` do this:

```
this.state.setState({someRoles: this.state.someRoles});
```

This lets React know that only ```someRoles``` has changed and allows React to optimize the rerendering, whereas ```forceUpdate``` forces React to assume that potentially all state has changed which inhibits some optimization opportunities. For this particular Component it's not big deal but it's better to get into the habit of doing it this way in general except for some rare circumstances.